app/src/org/commcare/android/resource/installers/ProfileAndroidInstaller.java

```diff
@@ -79,6 +87,7 @@ public boolean install(Resource r, ResourceLocation location, Reference ref,
             throws UnresolvedResourceException, UnfullfilledRequirementsException {
         //First, make sure all the file stuff is managed.
         super.install(r, location, ref, table, platform, upgrade, recovery);
+        storeReleasedTime(platform, ref);
```

Comment:
I'm not sure the statefulness of all of these elements actually supports this workflow quite right.

In the current code workflow, 'ref' only actually retains the released time headers as long as the existing object is functional. Sometimes super.install will take multiple attempts across multiple runs to actually complete, but the profile will only actually get fetched remotely on the very first attempt, meaning that state will likely be gone by the time we reach here. 

I think all of this lifecycle management needs to be moved to

https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/engine/resource/AndroidResourceManager.java#L108

(the class, not that exact line of code)

where we actually semantically know what's happening with the update process. That'll make it clearer where we are in the process and what we are tracking about the update itself.

Reply:
@ctsims Unfortunately I am having a hard time in imagining how this will work. Moving this code to `AndroidResourceManager` would mean that we will loose the access to the existing `ref` object with info about releasedOnTime. So are you suggesting that we store the `ref` object or make a static reference to`ref` from `AndroidResourceManager` ?

Reply:
Hey, you are right that this is harder than I thought. I forgot that we were still tracking the resource location in `AndroidResourceManager` as a string, and it only gets derived down to a reference through the IoC layer in `ResourceTable`. In order for my stated approach to make sense we'd have to change a bunch of code around so that the `ResourceLocation` was based on references (which I don't think is great).

After review, I think it's still not great that there's state sticking around inside of the reference object, but I can't think of a cleaner way to implement this than you have here, so let's keep the approach you are already using.